src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/DefaultCostCalculator.java

```diff
@@ -194,8 +204,9 @@ private int boardingCostConstrainedTransfer(
 
       int cost = waitFactor * boardWaitTime;
 
-      if (stopVisitCost != null) {
-        cost += stopVisitCost[boardStop];
+      // If it's first boarding event then it is not a transfer
+      if (stopTransferCost != null && !firstBoarding) {
+        cost += stopTransferCost[boardStop];
       }
```

Comment:
```suggestion
// StopTransferCost is NOT added to the cost here. This is because a trip-to-trip constrained transfer take
// precedence over stop-to-stop transfer priority. NeTEx station transfer priority.  
```